core/src/main/scala/org/apache/spark/CacheManager.scala

```diff
@@ -44,7 +44,14 @@ private[spark] class CacheManager(blockManager: BlockManager) extends Logging {
     blockManager.get(key) match {
       case Some(blockResult) =>
         // Partition is already materialized, so just return its values
+        val existingMetrics = context.taskMetrics.inputMetrics
+        val prevBytesRead = existingMetrics
+          .filter(_.readMethod == blockResult.inputMetrics.readMethod)
+          .map(_.bytesRead)
+          .getOrElse(0L)
```

Comment:
Yes, when the cache is being used then yes there will be clobbering. There are a few solutions
1) we can not filter on readMethod and just append blindly. That way we don't override any metrics but the eventual read method will not be correct (either first wins or last wins - whatever we choose)
2) we model input metrics like we do with shuffle metrics where we collect an array of them and then finally we sum them up. - this is a bigger change.